app/viewmodels/checkAnswers/PackagingSiteDetailsSummary.scala

```diff
@@ -0,0 +1,130 @@
+    answers.get(PackagingSiteDetailsPage).map {
+      answer =>
+
+        val value = if (answer) "site.yes" else "site.no"
+
+        SummaryListRowViewModel(
+          key = "packagingSiteDetails.checkYourAnswersLabel",
+          value = ValueViewModel(value),
+          actions = Seq(
+            ActionItemViewModel("site.change", routes.PackagingSiteDetailsController.onPageLoad(CheckMode).url)
+              .withVisuallyHiddenText(messages("packagingSiteDetails.change.hidden"))
+          )
+        )
+    }
+
+
+  def row2(packagingSiteList: List[Site])(implicit messages: Messages): List[SummaryListRow] = {
+    packagingSiteList.map {
+      site =>
+        ValueViewModel(
+          HtmlContent(
+            HtmlFormat.escape(site.address.lines.toString())
+          )
+        )
+        SummaryListRow(
+          key = Key(
+            content =
+              HtmlContent(addressFormatting(site)),
+            classes = "govuk-!-font-weight-regular govuk-!-width-two-thirds"
+          ),
+          actions = if (packagingSiteList.length > 1) {
+            Some(Actions("", Seq(
+              ActionItemViewModel("site.edit", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.hidden")),
+              ActionItemViewModel("site.remove", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.hidden"))
+            )))
+          } else {
+            Some(Actions("", Seq(
+              ActionItemViewModel("site.edit", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.edit.hidden"))
+            )))
+          }
+        )
+    }
+  }
+
+  private def addressFormatting(site: Site)(implicit messages: Messages): String = {
+    val commaFormattedSiteAddress = site.address.lines.map(line => { if (line.isEmpty) "" else line + ", " })
+
+    val addressNoTradingName = {
+      s"""${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val addressWithTradingName = {
+      s"""${HtmlFormat.escape(site.tradingName.get)}<br>${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}
+      ${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val separatePostCodeAddressNoTradingName = {
+      s"""${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}<br>${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val separatePostCodeAddressWithTradingName = {
+      s"""${HtmlFormat.escape(site.tradingName.get)}<br>${
+        HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))
+      }<br>${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val addressFormat = determineAddressFormat(site)
+
+    addressFormat match {
```

Comment:
Also, I think we should be using some kind of ENUM OR CASE OBJECT rather than string match, strings are brittle and I believe we have a warning whereby the match statement is not exhaustive
We can see its exhaustive but, things could change in the future.
https://docs.scala-lang.org/overviews/scala-book/case-objects.html